src/builder/QuarkBuilderBase.sol

```diff
@@ -407,25 +417,36 @@ contract QuarkBuilderBase {
         uint256 chainId,
         address account,
         uint256 blockTimestamp,
-        bool useQuotecall
+        bool useQuotecall,
+        bool useWrapUpTo
     ) internal pure {
         // Check if inserting wrapOrUnwrap action is necessary
         uint256 assetBalanceOnChain =
             Accounts.getBalanceOnChain(assetSymbol, chainId, chainAccountsList) + supplementalBalance;
         if (assetBalanceOnChain < amount && TokenWrapper.hasWrapperContract(chainId, assetSymbol)) {
-            // If the asset has a wrapper counterpart, wrap/unwrap the token to cover the transferIntent amount
+            // If the asset has a wrapper counterpart, wrap/unwrap the token to cover the amount needed for the intent
             string memory counterpartSymbol = TokenWrapper.getWrapperCounterpartSymbol(chainId, assetSymbol);
 
             // Wrap/unwrap the token to cover the amount
+            uint256 amountToWrap;
+            if (useWrapUpTo) {
+                // If we are using the "wrap up to script", then the `amountToWrap` should be the entire amount needed
+                // for the intent
+                amountToWrap = amount;
+            } else {
+                // If we aren't using the "wrap up to" script, then we need to subtract the current balance from the
```

Comment:
I'm trying to understand what's the serious case to not use `wrapUpTo`. Isn't it basically just a superior version of the same intent here?

Reply:
In most cases, we know exactly how much needs to be wrapped so we don't need the extra overhead of `wrapUpTo`. But thinking about this again, that statement's not true. There can always be transactions happening outside of this Quark Operation that modifies the user's balance prior to execution of this operation. For this reason, I think we should just always default to `wrapUpTo`.

Reply:
Updated to always use `wrap/unwrap all`